plugin/src/resolve-gatsby-image-data.ts

```diff
@@ -4,17 +4,51 @@ import {
   IImage,
   ImageFormat,
 } from "gatsby-plugin-image";
+import fetch from "node-fetch"
 import { ShopifyImage, urlBuilder } from "./get-shopify-image";
+import sharp from "sharp"
 
 type ImageLayout = "constrained" | "fixed" | "fullWidth";
 
+type IImageWithPlaceholder = IImage & {
+  placeholder: string
+}
+
+/**
+ * Get the URL for a 20-px wide image to use as a blurred placeholder
+ * 
+ * The image must be downloaded and converted to a base64-encoded data URI
+ * 
+ * @param imageArgs 
+ */
+async function getLowResolutionImageBase64(imageArgs: Node & { width: number; height: number; originalSrc: string }): Promise<string> {
+  const response = await fetch(imageArgs.originalSrc)
+  const buffer = await response.buffer();
+  const resizedBuffer = await sharp(buffer)
+        .resize({
+          fit: sharp.fit.inside,
+          width: 20
+        })
+        .toBuffer()
+  return resizedBuffer.toString(`base64`)
+}
+
+/**
+ * Download and generate a low-resolution placeholder
+ * 
+ * @param lowResImageFile
+ */
+function getBase64DataURI({ imageBase64 }: { imageBase64: string }) {
+  return `data:image/png;base64,${imageBase64}`
+}
+
 export async function resolveGatsbyImageData(
   image: Node & ShopifyImage,
   {
     formats = ["auto"],
     layout = "constrained",
-    ...options
-  }: { formats: Array<ImageFormat>; layout: ImageLayout }
+    options
+  }: { formats: Array<ImageFormat>; layout: ImageLayout; options: { placeholder: string } }
```

Comment:
This looks like a potential mistake to me... why did you remove the spread operator?

Reply:
@sslotsky this was unintentional - removing :+1:

Reply:
Ah, I think I _had_ added this, but to avoid this type error:
> `Property 'placeholder' does not exist on type '{ options: { placeholder: string; }; }'.`

After reverting, the type error is once again present, but it seems we can ignore it; the placeholder appears to be getting put in place as expected.